### PR TITLE
Warn when message rate exceeds limit for RateLimiter

### PR DESCRIPTION
**Pull request by Rustinante**
## Motivation
Addressing the warning feature from #3255

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: common/stream-ratelimiter/src/lib.rs [hunk 1]
```diff
@@ -76,6 +77,12 @@ impl<T: Stream> Stream for RateLimiter<T> {
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut task::Context<'_>) -> Poll<Option<Self::Item>> {
         // If the interval is ready, add permits. This is safe to do in all calls to poll_next().
         if let Poll::Ready(Some(_)) = self.as_mut().project().interval.poll_next(cx) {
+            if self.permits.available_permits() == 0 {
```
Comment (davidiw) [inline common/stream-ratelimiter/src/lib.rs:80]: I didn't review this code, but why this here versus at line 84/91 where we didn't acquire a permit?

**Comment by davidiw** [comment]
Thanks for the PR, please add appropriate reviewers (myself and @phlip9) for networking, so we get notified :)

File: common/stream-ratelimiter/src/lib.rs (hunk 1 above)
Comment (phlip9) [reply common/stream-ratelimiter/src/lib.rs:80]: didn't review this code either... also I think L100 (not from this PR) is a memory leak 😬 

tbh not totally sure this code is correct...

Comment (phlip9) [reply common/stream-ratelimiter/src/lib.rs:80]: one issue with this warn log is we have no context here (namely peer_id)

**Comment by Rustinante** [comment]
Thanks, let me actually convert this PR into a draft first. I'll focus on the task of refactoring the `NetworkBuilder`, and the come back to this.

**Comment by davidiw** [comment]
I think what we actually want to do here is strip this code out of the code base, I don't think it provides us much protection at all and gives a false sense of security. Feel free to follow up :)

**Pull request closed without merging**